xbmc/cores/dvdplayer/DVDPlayer.cpp

```diff
@@ -4705,7 +4705,6 @@ bool CDVDPlayer::SwitchChannel(const CPVRChannelPtr &channel)
 bool CDVDPlayer::CachePVRStream(void) const
 {
   return m_pInputStream->IsStreamType(DVDSTREAM_TYPE_PVRMANAGER) &&
-      (!g_PVRManager.IsPlayingRecording() ||
-          (m_item.HasPVRRecordingInfoTag() && m_item.GetPVRRecordingInfoTag()->IsBeingRecorded()))&&
+      (!g_PVRManager.IsPlayingRecording() || g_PVRManager.IsPlayingRecordingInProgress()) &&
```

Comment:
this looks complicated and goes in wrong direction. actually we want buffering for live streams so a simple IsLiveStream would fit best. This can even change for a recording in progress because buffering is only required at and of file when it's live.